### DISCUSSION TITLE
Issue when converting odt to md (LibreOffice + Zotero -> Zettlr + CSL Json through BBT)

### DISCUSSION THREAD

**Discussion by artav4** [retorquere/zotero-better-bibtex#3016] [Q&A] [answered]
Hello, sorry in advance but I didn't know where to ask and if it is the appropriate place.

I want to convert an odt file containing Zotero citations to a markdown file for use in Zettlr. I have installed BetterBibTex in Zotero and followed those docs: 
[BBT doc](https://retorque.re/zotero-better-bibtex/citing/migrating/index.html)
[Zettlr doc](https://retorque.re/zotero-better-bibtex/citing/migrating/index.html)

<details><summary>However, when I import a file (odt or already converted md), the citations look like this:</summary>

```
[]{#ZOTERO_ITEM CSL_CITATION {\"citationID\":\"Yox0QaH2\",\"properties\":{\"formattedCitation\":\"(Thierry et al., 1995)\",\"plainCitation\":\"(Thierry et al., 1995)\",\"noteIndex\":0},\"citationItems\":[{\"id\":457,\"uris\":[\"http://zotero.org/users/8907875/items/AUZAT3TI\"],\"itemData\":{\"id\":457,\"type\":\"article-journal\",\"abstract\":\"This article examines strategic production and operations management issues in product recovery management (PRM). PRM encompasses the management of all used and discarded products, components, and materials for which a manufacturing company is legally, contractually, or otherwise responsible. The objective of PRM is to recover as much of the economic (and ecological) value of used and discarded products, components, and materials as reasonably possible, thereby reducing the ultimate quantities of waste to a minimum. This article also discusses the relevance of PRM to durable products manufacturers. It contains a categorization of PRM decisions. A case study based on the PRM system of a multinational copier manufacturer is presented to illustrate a set of specific production and operations management issues. The experiences of two other pro-active manufacturers (BMW and IBM) are also discussed.\",\"container-title\":\"California Management Review\",\"DOI\":\"10.2307/41165792\",\"ISSN\":\"00081256\",\"issue\":\"2\",\"note\":\"publisher: California Management Review\",\"page\":\"114-135\",\"source\":\"EBSCOhost\",\"title\":\"Strategic Issues in Product Recovery Management: California Management Review\",\"title-short\":\"Strategic Issues in Product Recovery Management\",\"volume\":\"37\",\"author\":[{\"family\":\"Thierry\",\"given\":\"Martijn\"},{\"family\":\"Salomon\",\"given\":\"Marc\"},{\"family\":\"Van Nunen\",\"given\":\"Jo\"},{\"family\":\"Van Wassenhove\",\"given\":\"Luk\"}],\"issued\":{\"date-parts\":[[\"1995\"]],\"season\":\"Winter\"}}}],\"schema\":\"https://github.com/citation-style-language/schema/raw/master/csl-citation.json\"} RNDjH0j24CkLG}(Thierry et al., 1995)
```

</details>

I have absolutely no idea why the output is like that but if you have any clue let me know

**Comment by retorquere** [comment]
You'll have to ask the pandoc people -- while the `docx+citations` pandoc command is running, BBT is not involved at all.

Was this document produced before BBT was installed? the `id` values are Zotero item keys, not citekeys, and references inserted with BBT active would have had citation keys. That's why the conversion using just pandoc works.

**Reply by artav4** [reply]
Hello @retorquere, thank you for the fast answer, the document is an odt so I cannot use the `docx+citations` command, unless I convert the document (I tried it and it seems that it was okay when opening with LO Writer, but as I am using Linux I cannot try with Word, and the conversion did not work).

And yes, the document was produced before BBT was installed.

> the id values are Zotero item keys, not citekeys, and references inserted with BBT active would have had citation keys.

I'm sorry but I did not understand that part of your answer, when I look at the citation there is a `citationID` as well as a `citation-key`.

The conversion in Pandoc gave me the same result as Zettlr import a files using Pandoc. If you cannot help me further or if it is not relevant anymore, just tell me and I will go ask the Pandoc folks!

**Reply by retorquere** [reply]
> Hello @retorquere, thank you for the fast answer, the document is an odt so I cannot use the `docx+citations` command, unless I convert the document (I tried it and it seems that it was okay when opening with LO Writer, but as I am using Linux I cannot try with Word, and the conversion did not work).

`odt+citations` might work. `pandoc --list-input-formats` does not list `docx+citations` so maybe it's just `<supported input format>+citations`.
 
> And yes, the document was produced before BBT was installed.

Then the information won't be there, so this

> If you install [this](https://retorque.re/zotero-better-bibtex/citing/migrating/better-bibtex-citekeys.csl) CSL style in Zotero (or modify it further and use that), Zotero will render the in-text citations as [@citekey] when you ask Zotero to render the bibliography. If you want LaTeX, modify the style accordingly. You can then put your document through pandoc or whatnot to get LaTeX/Markdown.

is the option you want to try. After you see the citation keys in the text, ask pandoc to convert it to markdown.

> I'm sorry but I did not understand that part of your answer, when I look at the citation there is a `citationID` as well as a `citation-key`.

I don't see `citation-key` in the output you posted above.

> The conversion in Pandoc gave me the same result as Zettlr import a files using Pandoc. If you cannot help me further or if it is not relevant anymore, just tell me and I will go ask the Pandoc folks!

If the document doesn't have the citation-key fields, the pandoc people can't help you. It may be possible to refresh the references in the docx so the BBT-generated keys are picked up, and then the pure-pandoc route could work (and it still is the best way).

**Reply by artav4** [reply]
> `odt+citations` might work. `pandoc --list-input-formats` does not list `docx+citations` so maybe it's just `<supported input format>+citations`.
No, it does not work, the `+citations` part of a pandoc command is what they call an extension, you can use `pandoc --list-extensions=format`, and it is listed for docx but not for odt unfortunately.
 
For the rest of the answer I double-checked and yes indeed in the example that I gave you, there is no `citation-key`. So I did install BBT and the CSL style, modify it to remove the brackets around a citation, apply it in my odt file and try to use pandoc. 

<details><summary>The output for the same citation is:</summary>

```
> []{#ZOTERO_ITEM CSL_CITATION {\"citationID\":\"Yox0QaH2\",\"properties\":{\"formattedCitation\":\"@thierryStrategicIssuesProduct1995\",\"plainCitation\":\"@thierryStrategicIssuesProduct1995\",\"noteIndex\":0},\"citationItems\":[{\"id\":72,\"uris\":[\"http://zotero.org/users/8907875/items/AUZAT3TI\"],\"itemData\":{\"id\":72,\"type\":\"article-journal\",\"abstract\":\"This article examines strategic production and operations management issues in product recovery management (PRM). PRM encompasses the management of all used and discarded products, components, and materials for which a manufacturing company is legally, contractually, or otherwise responsible. The objective of PRM is to recover as much of the economic (and ecological) value of used and discarded products, components, and materials as reasonably possible, thereby reducing the ultimate quantities of waste to a minimum. This article also discusses the relevance of PRM to durable products manufacturers. It contains a categorization of PRM decisions. A case study based on the PRM system of a multinational copier manufacturer is presented to illustrate a set of specific production and operations management issues. The experiences of two other pro-active manufacturers (BMW and IBM) are also discussed.\",\"container-title\":\"California Management Review\",\"DOI\":\"10.2307/41165792\",\"ISSN\":\"00081256\",\"issue\":\"2\",\"note\":\"publisher: California Management Review\",\"page\":\"114-135\",\"source\":\"EBSCOhost\",\"title\":\"Strategic Issues in Product Recovery Management: California Management Review\",\"title-short\":\"Strategic Issues in Product Recovery Management\",\"volume\":\"37\",\"author\":[{\"family\":\"Thierry\",\"given\":\"Martijn\"},{\"family\":\"Salomon\",\"given\":\"Marc\"},{\"family\":\"Van Nunen\",\"given\":\"Jo\"},{\"family\":\"Van Wassenhove\",\"given\":\"Luk\"}],\"issued\":{\"date-parts\":[[\"1995\"]],\"season\":\"Winter\"},\"citation-key\":\"thierryStrategicIssuesProduct1995\"}}],\"schema\":\"https://github.com/citation-style-language/schema/raw/master/csl-citation.json\"} RNDLgSrbHjnY5}\@thierryStrategicIssuesProduct1995
```
</details>

Is that a normal output? Then if it the case, it would be my Zettlr that have a hard time re-styling citations.

**Reply by retorquere** [reply]
That is not the expected output; the style I linked to has

```
<layout delimiter="; " prefix="[" suffix="]">
      <group>
        <text value="@"/>
        <text variable="citation-key"/>
        <text variable="locator" prefix=", p. "/>
      </group>
    </layout>
```

so should output pandoc-ish references. I don't have any experience with using CSL though; someone wrote that style for me a long time ago, and I just posted it. I don't use Word/LO unless someone forces me to. People on the Zotero forums will be of more help.

**Reply by artav4** [reply]
That's the weird thing! In the odt file, the citations have the good style (something like [@nametitleyear]) but the weird citations that I have in my md files/zettlr seem to be caused by the conversion with pandoc.

I thought about unlinking references in the odt file before the conversion but wouldn't that mean that the citations won't work in zettlr because the citation keys would be lost?

**Reply by retorquere** [reply, marked as answer]
Answers crossed. Unlinking means it becomes just plain text, but because that plain text is `@citationkey`, it will be picked up as a citation by Zettlr/pandoc.

**Reply by artav4** [reply]
Exactly, that worked, thanks a lot for your help, I will stay away from using writer when I need to reference something!
